### PR TITLE
Add quill database access library

### LINKED ISSUE #421: Add option for Quill database access
We want users of the template to have a choice of database integration library. Right now it's only Slick. This ticket is about introducing Quill.
- add Quill implementation of ItemRepository (named QuillItemRepository) and HealthCheckService (named QuillHealCheckService) in the infrastucture module
- add in-memory implementation of ItemRepository (named InMemoryItemRepository) and HealthCheckService (named InMemoryHealCheckService) in the infrastucture module
- add a choice between Slick and Quill implementation when running the template (analogous to Akka HTTP vs ZIO HTTP choice)
- if the user answers "no" to both Slick and Quill question, then the in-memory implementation should be used

### PR DESCRIPTION
**Pull request by edrzmr**
Closes #421

### CODE REVIEW

File: src/main/g8/default.properties [hunk 1]
```diff
@@ -1,7 +1,9 @@
-name=zio-akka-quickstart
+name=zio-scala2-quickstart
 description=This is a seed project which creates ready to use Scala application using ZIO, Akka HTTP and Slick.
```
Comment (jczuchnowski) [inline src/main/g8/default.properties:2]: It's about time to update this description.

File: src/main/g8/src/main/scala/$package$/Layers.scala [hunk 2]
```diff
@@ -1,22 +1,78 @@
...
 
+$if(enable_slick.truthy)$
+  object Repository {
+    import com.typesafe.config.Config
+    import slick.interop.zio.DatabaseProvider
+    import slick.jdbc.JdbcProfile
+    import slick.jdbc.PostgresProfile
+    import $package$.infrastructure.slick.{ SlickHealthCheckService, SlickItemRepository }
+    import $package$.api.healthcheck.HealthCheckService
+    import $package$.domain.ItemRepository
```
Comment (jczuchnowski) [inline src/main/g8/src/main/scala/$package$/Layers.scala:24]: Can we make these a top level imports? It's fine if they are enclosed in another `enable_slick.truthy` test.

File: src/main/g8/src/main/scala/$package$/Layers.scala [hunk 3]
```diff
@@ -1,22 +1,78 @@
...
+  object Repository {
+    import com.typesafe.config.Config
+    import com.typesafe.config.ConfigFactory
+    import io.getquill.jdbczio.Quill
+    import io.getquill.Literal
+    import javax.sql.DataSource
+    import scala.jdk.CollectionConverters.MapHasAsJava
+    import $package$.infrastructure.quill.{QuillHealthCheckService, QuillItemRepository}
+    import $package$.api.healthcheck.HealthCheckService
+    import $package$.domain.ItemRepository
```
Comment (jczuchnowski) [inline src/main/g8/src/main/scala/$package$/Layers.scala:52]: Same suggestion as for Slick.

File: src/main/g8/src/main/scala/$package$/Layers.scala (hunk 2 above)
Comment (edrzmr) [reply src/main/g8/src/main/scala/$package$/Layers.scala]: done

File: src/main/g8/default.properties (hunk 1 above)
Comment (edrzmr) [reply src/main/g8/default.properties]: done

File: src/main/g8/src/main/scala/$package$/Layers.scala (hunk 3 above)
Comment (edrzmr) [reply src/main/g8/src/main/scala/$package$/Layers.scala]: done

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**